src/Engine/MapController.java

```diff
@@ -221,6 +226,7 @@ private void handleFreeTileSelect(InputHandler.InputAction input)
               threats.remove(resident);
             else
               threats.add(resident);
+            OverlayCache.instance(myGame).InvalidateCache();
```

Comment:
When in `THREATS` mode, flagging/unflagging units works immediately. When in any other mode, it only takes effect once the mode changes again. This is really only obvious because of the `T` mark that gets applied. Strange nonetheless.

Reply:
The flagging goes through immediately; the mark doesn't show up because the overlay didn't change.
FWIW, I kinda wanted to put `OverlayCache` on the UI side of things, but that would prevent `MapController` from directly notifying it when the list changes. (there are ways around that, but I'm not super worried about it at this point)